lib/routes/account.js

```diff
@@ -581,41 +578,54 @@ module.exports = function (
         }
 
         function checkEmailAndPassword () {
-          // Session token verification is only enabled for certain users during phased rollout.
-          //
-          // If the user went through the sigin-unblock flow, they have already verified their email.
-          // No need to also require confirmation afterwards.
-          //
-          // Even when it is enabled, we only do the email challenge if:
-          //  * the request wants keys, since unverified sessions are fine to use for e.g. oauth login.
-          //  * the email is verified, since content-server triggers a resend of the verification
-          //    email on unverified accounts, which doubles as sign-in confirmation.
-          //  * the login is flagged that it can be bypassed
-
-          // Check to see if this login can bypass sign-in confirmation. Current scenarios include
-          //  * User has already logged in from this ip address and verified the sign-in
-          let bypassSiginConfirmation = features.canBypassSiginConfirmation(emailRecord.email, securityEventVerified, securityEventRecency)
-          if (bypassSiginConfirmation) {
-            log.info({
-              op: 'Account.ipprofiling.seenAddress',
-              uid: emailRecord.uid.toString('hex')
-            })
+          // All sessions are considered unverified by default.
+          needsVerificationId = true
+
+          // However! To help simplify the login flow, we can use some heuristics to
+          // decide whether to consider the session pre-verified.  Some accounts
+          // get excluded from this process, e.g. testing accounts where we want
+          // to know for sure what flow they're going to see.
+
+          var forceSigninConfirmation = false
+          if (signinConfirmation.forcedEmailAddresses) {
+            if (signinConfirmation.forcedEmailAddresses.test(emailRecord.email)) {
+              forceSigninConfirmation = true
+            }
           }
 
-          if (didSigninUnblock || !features.isSigninConfirmationEnabledForUser(emailRecord.uid, emailRecord.email, request)
-            || bypassSiginConfirmation) {
+          if (! forceSigninConfirmation) {
+
+            // If they're logging in from an IP address on which they recently did
+            // another, successfully-verified login, then we can consider this one
+            // verified as well without going through the loop again.
+            if (features.isSecurityHistoryProfilingEnabled()) {
+              if (securityEventVerified && securityEventRecency === 'day') {
+                log.info({
+                  op: 'Account.ipprofiling.seenAddress',
+                  uid: emailRecord.uid.toString('hex')
+                })
+                needsVerificationId = false
+              }
+            }
```

Comment:
I realized that I missed something here - the handling of "suspicious" requests.  I'm going to go ahead and create two helper functions "forceSigninConfirmation" and "skipSigninConfirmation" into which we can pull the necessary logic.